File: 169-majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let fr = 0;
    let ans = 0;

    for(let i=0; i<nums.length; i++){
        if(fr == 0){
            ans = nums[i];
        }
        if(ans == nums[i]){
            fr++;
        }else{
            fr--;
        }
    }
    return ans;
};

File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        ans=nums[0]
        count=0
        for i in range(len(nums)):
            if(count == 0):
                ans = nums[i]
            if(ans != nums[i]):
                count-=1
            else:
                count+=1
        return ans
    
            
                
        